### PR TITLE
Add JsonMetadataTable in preparation for batch table refactor

### PR DESCRIPTION
**Pull request by ptrgags**
This branch targets `3d-tiles-next`.

This adds a `JsonMetadataTable`, a class that serves a similar purpose as `MetadataTable` does, though instead of binary properties, it stores free-form JSON arrays. This is for compatibility with [Batch Tables](https://github.com/CesiumGS/3d-tiles/tree/master/specification/TileFormats/BatchTable).

This PR does not handle batch table hierarchy or parsing batch tables, that will be in future PRs

@lilleyse

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ptrgags!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

File: Source/Scene/JsonMetadataTable.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import DeveloperError from "../Core/DeveloperError.js";
+import MetadataEntity from "./MetadataEntity.js";
+
+/**
+ * A table for storing free-form JSON metadata, as in the 3D Tiles batch table.
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {Number} options.count The number of entities in the table.
+ * @param {Object.<String, Array>} options.properties The JSON representation of the metadata table. All the arrays must have exactly options.count elements.
+ * @param {MetadataClass} [options.class] The class that group metadata conforms to.
```
Comment (lilleyse) [inline Source/Scene/JsonMetadataTable.js:13]: Should `options.class` be removed from the doc?

File: Source/Scene/JsonMetadataTable.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ * @returns {*} The value of the property or <code>undefined</code> if the property does not exist.
+ * @exception {DeveloperError} index is out of bounds
+ * @private
+ */
+JsonMetadataTable.prototype.getProperty = function (index, propertyId) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.number("index", index);
+  Check.typeOf.string("propertyId", propertyId);
+
+  if (index < 0 || this._count < index) {
```
Comment (lilleyse) [inline Source/Scene/JsonMetadataTable.js:64]: The code above is off by one, should be `this._count <= index`. Personally I find it more readable like
```suggestion
  if (index < 0 || index >= this._count) {
```

which is what `MetadataTableProperty.js` does

File: Source/Scene/JsonMetadataTable.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    throw new DeveloperError("propertyId " + propertyId + " does not exist");
+  }
+
+  if (index < 0 || this._count < index) {
+    throw new DeveloperError("index must be in [0, " + this._count + ")");
+  }
+  //>>includeEnd('debug');
+
+  var property = this._properties[propertyId];
+  if (defined(property)) {
```
Comment (lilleyse) [inline Source/Scene/JsonMetadataTable.js:102]: The property should be defined, otherwise a DeveloperError would have been thrown.

**Comment by ptrgags** [comment]
@lilleyse updated!

**Comment by lilleyse** [comment]
Thanks!

**Pull request merged**